backend/inspirehep/legacy/views.py

```diff
@@ -46,54 +50,80 @@ def _redirect_author(bai):
 
 @blueprint.route("/author/claim/<bai>", methods=("GET",))
 def redirect_claim(bai):
-    return redirect(f"{current_app.config['LEGACY_BASE_URL']}/author/claim/{bai}", 302)
+    if current_app.config["FEATURE_FLAG_ENABLE_LEGACY_VIEW_REDIRECTS"]:
+        return redirect(
+            f"{current_app.config['LEGACY_BASE_URL']}/author/claim/{bai}", 302
+        )
+
+    recid = get_pid_for_pid("bai", bai, "recid")
+    return redirect(f"/authors/{recid}", 302)
 
 
 @blueprint.route("/author/merge_profiles", methods=("GET",))
 def redirect_merge_profiles():
-    return redirect(
-        f"{current_app.config['LEGACY_BASE_URL']}/author/merge_profiles?{request.query_string.decode('utf-8')}",
-        302,
-    )
+    if current_app.config["FEATURE_FLAG_ENABLE_LEGACY_VIEW_REDIRECTS"]:
+        return redirect(
+            f"{current_app.config['LEGACY_BASE_URL']}/author/merge_profiles?{request.query_string.decode('utf-8')}",
+            302,
+        )
+
+    return redirect(f"/authors", 302)
 
 
 @blueprint.route("/author/manage_profile/<bai>", methods=("GET",))
 def redirect_manage_profile(bai):
-    return redirect(
-        f"{current_app.config['LEGACY_BASE_URL']}/author/manage_profile/{bai}", 302
-    )
+    if current_app.config["FEATURE_FLAG_ENABLE_LEGACY_VIEW_REDIRECTS"]:
+        return redirect(
+            f"{current_app.config['LEGACY_BASE_URL']}/author/manage_profile/{bai}", 302
+        )
+    recid = get_pid_for_pid("bai", bai, "recid")
+    return redirect(f"/authors/{recid}", 302)
 
 
 @blueprint.route("/search", methods=("GET",))
 def redirect_query():
-    # TODO: Delete redirection to legacy when Institutions and Experiments are
-    # on labs and legacy UI is completely shut down
-
     legacy_collection = request.args.get("cc", "HEP")
     query = request.args.get("p", "")
 
     collection = current_app.config["COLLECTION_EQUIVALENCE"].get(legacy_collection)
     if collection:
         return redirect(f"/{collection}?q={query}", 301)
 
-    return redirect(
-        f"{current_app.config['LEGACY_BASE_URL']}/search?{request.query_string.decode('utf-8')}",
-        302,
+    if current_app.config["FEATURE_FLAG_ENABLE_LEGACY_VIEW_REDIRECTS"]:
+        return redirect(
+            f"{current_app.config['LEGACY_BASE_URL']}/search?{request.query_string.decode('utf-8')}",
+            302,
+        )
+
+    collection = MAP_LEGACY_COLLECTION_TO_HIDDEN_COLLECTION.get(
+        legacy_collection.lower()
     )
```

Comment:
sorry, maybe I wasn't clear in zulip. I don't believe this is needed, as we've already aligned the metadata in `_collections` with the legacy collection name in dojson. So you can do (case normalization is already handled by ES, so not needed either).
```suggestion
    collection = legacy_collection
```